### DISCUSSION TITLE
Debian 11 - Backups and Invoke

### DISCUSSION THREAD

**Discussion by imager2019** [inventree/InvenTree#5263] [General]
Sorry if these questions are very basic in nature, but I am very new to Linux (Debian 11).

Have a full install running on Debian 11 with just over 2,100 items migrated in from inFlow.

Want to make sure that I have good backups in place.  I have enabled Automatic Backups (admin...Settings...Server).

![image](https://github.com/inventree/InvenTree/assets/49574724/4df5c371-c4b7-4487-9d43-3d32b5364cb2)

I am, however, not seeing the backups. Looking in  opt...inventree...data...backup  . I do see backups from Friday and Saturday but not Sunday or Monday. Is there something else I need to enable/perform for the backups to occur, or maybe I am watching in the wrong location for them?

I have reviewed (and understood maybe 20%) of https://docs.inventree.org/en/latest/start/backup/#storage-providers

I have also tried from the Debian Terminal the:  **invoke backup** command, but receive a command not found response
![image](https://github.com/inventree/InvenTree/assets/49574724/b955cf7e-d040-4319-886c-79811966a8e6)

Did I not install everything I needed to when I setup Debian or maybe running the command the wrong way?

Appreciate any thoughts.

john

**Comment by wolflu05** [comment]
Hello @imager2019 ,

you first need to source the venv. To do so, search for a folder `venv` or `env` that should contain a folder called `bin` and `lib`. Then execute the following (replace venv with whatever your folder is named): 
```bash
source ./venv/bin/activate
```
Then you should be able to run the `invoke backup` command.

**Reply by imager2019** [reply]
Thanks for the feedback...
Found a `venv` folder, located at `usr/lib/python3.9` . It only contains two folders (`__pycashe__` and `scripts`) and two files (`__init__.py` and `__main__.py`) - no `bin` or `lib`. The scripts folder does contain a `scripts..common` folder which does have an activate file in it.

I did find an `env` folder (under `opt/inventree/env`) which does contain bin and lib (along with` include, lib64, share folders and pyvenv.cfg`)

From the `opt/inventree` folder, I enter the `source ./venv/bin/activate` command, and did not receive an error, just back to the command prompt. I than entered  `invoke backup` and receive the following, which I believe indicates the backup failed.

![image](https://github.com/inventree/InvenTree/assets/49574724/86ee3ac2-630e-4234-97df-511618020012)

Am I missing something along the way??
